tests/cases/conformance/types/typeRelationships/typeInference/genericContextualTypes2.ts

```diff
@@ -0,0 +1,41 @@
+// @strict: true
```

Comment:
You can play around with those test cases in [the TS 4.7 playground](https://www.typescriptlang.org/play?ts=4.7.4#code/C4TwDgpgBAMg9gdwJIDsBmEBOAeAKgPigF4pcoAyKAbwF8BuAKAdEigAUBDTYASw4BsAggGdhPAOYoseAMJwUwCAA9gAGlIBpCCCjLFKACbCoAa21w0pOQr2ESACgZQoAY3mKVALivu9DAJTEhLjWHsAA2rhaIAC6jMzg0GyYcJDcICJiktIhvip21E5Q4RpQPCim5pa5NioxAPzenNx8QqISUjg1YeoahAA+PrURGnEM9Ews0ACyEMAcsnnABVRFHC688t72G0re3XqBRIQAbnA8BuNM5YqYaOvQmR2CGzzyi8OEq85T3gBESmE80UADoOO1JH9GM4APQwqD2LjifahPTqAC2cw43lm8w+YXw-m8Zwu0N0SggLm2SJRSwxWJxWPxtiOp3OBkYNCYBkp-C40DQAFcUK95FBwVkUMz8o5nBKOpiFE0UmlQE9sjh4Mh0DlUfl8AFvOqUC9NlKDvl4lMoNY0BJpctiIVnG5hrThmSIApMCBGlBjaa3ua9ctOfEeS4+ZgBcLRRUXNGOIppusABblCAO-A7eR25E23P2i3LIlQEkcpgJiBJiAplzpqTYKiuODC4DeFCC9EAIywUBo2e+LbdzucLbb3gADKoijQZ84vcAfd55ZJ7EOXa2lQjdt5m24J1BOz2+zRWVAANQX3Ygg8Kef9-wzs+MIA) where they error.